File: markov/src/python/createMarkov.py
"""
CSV読み込みのコードをここに書く
"""

data = [[]]  # CSVデータを格納するリスト(2次元配列)

gen = 1  # 何つ前まで見るか

chain = {}

for sentence in data:
    sentence = ["[BOS]"] + sentence + ["[EOS]"]
    for i in range(len(sentence) - gen):
        if "".join(sentence[i:i + gen]) not in chain:
            chain["".join(sentence[i:i + gen])] = []
        chain["".join(sentence[i:i + gen])].append(sentence[i + gen])

File: markov/src/ts/createMarkov.ts
/*
CSV読み込みのコードをここに書く
*/

let data: string[][] = [[]];  // CSVデータを格納するリスト(2次元配列)

let gen: number = 1;  // 何つ前まで見るか

let chain: { [key: string]: string[] } = {};

for (let sentence of data) {
  sentence = ["[BOS]", ...sentence, "[EOS]"];
  for (let i = 0; i < sentence.length - gen; i++) {
    const key = sentence.slice(i, i + gen).join("");
    if (!(key in chain)) {
      chain[key] = [];
    }
    chain[key].push(sentence[i + gen]);
  }
}
